controllers/snapshot/snapshot_adapter.go

```diff
@@ -147,6 +153,100 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (reconciler.Operation
 	return reconciler.ContinueProcessing()
 }
 
+// EnsureCreationOfEnvironment makes sure that all envrionments that were requested via
+// IntegrationTestScenarios get created, in case that environment is already created, provides
+// a message about this fact
+func (a *Adapter) EnsureCreationOfEnvironment() (reconciler.OperationResult, error) {
+	if gitops.HaveHACBSTestsFinished(a.snapshot) {
+		a.logger.Info("The Snapshot has finished testing.")
+		return reconciler.ContinueProcessing()
+	}
+
+	integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+
+	if err != nil {
+		a.logger.Error(err, "Failed to get Integration test scenarios for the following application",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return reconciler.RequeueOnErrorOrStop(err)
+	}
+
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		a.logger.Error(err, "Failed to get all environments.",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return reconciler.RequeueOnErrorOrStop(err)
+	}
+
+	environmentFound := false
+	if integrationTestScenarios != nil {
+		for _, integrationTestScenario := range *integrationTestScenarios {
+			integrationTestScenario := integrationTestScenario //G601
+			if reflect.ValueOf(integrationTestScenario.Spec.Environment).IsZero() {
+				continue
+			}
+			for _, environment := range *allEnvironments {
+				environment := environment //G601
+				//prevent creating already existing environments
+				if helpers.HasLabelWithValue(&environment, gitops.SnapshotLabel, a.snapshot.Name) && helpers.HasLabelWithValue(&environment, gitops.SnapshotTestScenarioLabel, integrationTestScenario.Name) {
+					a.logger.Info("Environment already exists and contains snapshot and scenario:",
+						"Environment.Name", environment.Name,
+						"Snapshot.Name", a.snapshot.Name,
+						"IntegrationScenario.Name", integrationTestScenario.Name)
+					environmentFound = true
+					break
+				}
+			}
+
+			if environmentFound {
+				environmentFound = false
+				continue
+			}
```

Comment:
```suggestion
                TestScenarioLoop:
		for _, integrationTestScenario := range *integrationTestScenarios {
			integrationTestScenario := integrationTestScenario //G601
			if reflect.ValueOf(integrationTestScenario.Spec.Environment).IsZero() {
				continue
			}
			for _, environment := range *allEnvironments {
				environment := environment //G601
				//prevent creating already existing environments
				if helpers.HasLabelWithValue(&environment, gitops.SnapshotLabel, a.snapshot.Name) && helpers.HasLabelWithValue(&environment, gitops.SnapshotTestScenarioLabel, integrationTestScenario.Name) {
					a.logger.Info("Environment already exists and contains snapshot and scenario:",
						"Environment.Name", environment.Name,
						"Snapshot.Name", a.snapshot.Name,
						"IntegrationScenario.Name", integrationTestScenario.Name)
					continue TestScenarioLoop
				}
			}
```

Reply:
We don't need `environmentFound` variable in this way

Reply:
Great usage, learned it. Thanks!